examples/tableau/src/components/dashboard.tsx

```diff
@@ -0,0 +1,247 @@
+import quip from "quip-apps-api";
+import React, {ChangeEvent, useEffect, useRef, useState} from "react";
+import {TABLEAU_BASE_URL, TABLEAU_JS_LIB} from "../config";
+import {AppData, FilterType, RootEntity} from "../model/root";
+import Dialog from "./dialog";
+import FilterManager from "./filters/filterManager";
+import ParamManager from "./parameters/paramManager";
+
+interface DashboardProps {
+    rootRecord: RootEntity;
+}
+
+const Dashboard = ({rootRecord}: DashboardProps) => {
+    const [data, setData] = useState<AppData>(rootRecord.getData());
+
+    const refreshData_ = () => {
+        const data = rootRecord.getData();
+        setData(data);
+    };
+
+    useEffect(() => {
+        // on mount
+        rootRecord.listen(refreshData_);
+        refreshData_();
+
+        return () => {
+            // on unmount
+            rootRecord.unlisten(refreshData_);
+            quip.apps.clearEmbeddedIframe();
+        };
+    }, []);
+
+    useEffect(() => {
+        // on mount
+        const script = document.createElement("script");
+        script.src = TABLEAU_JS_LIB;
+        document.body.appendChild(script);
+
+        return () => {
+            // on unmount
+            document.body.removeChild(script);
+        };
+    }, []);
+
+    const openSelectDashboard = () => {
+        rootRecord.openSelectDashboard();
+    };
+
+    const closeSelectDashboard = () => {
+        rootRecord.closeSelectDashboard();
+    };
+
+    const selectDashboard = () => {
+        rootRecord.setNewDashboard();
+    };
+
+    const updateNewDashboardUrl = (event: ChangeEvent<HTMLInputElement>) => {
+        rootRecord.setNewDashboardUrl(event.currentTarget.value.trim());
+    };
+
+    const vizContainer = useRef<HTMLDivElement | null>(null);
+
+    const filterChangedHandler_ = async (e) => {
+        const worksheet = e.detail.worksheet;
+        const fieldName = e.detail.fieldName;
+        console.log("Filter changed", e);
+        const filters = await worksheet.getFiltersAsync();
+        const filter = filters.find((f) => f.fieldName === fieldName);
+        if (filter) {
+            // rootRecord.setFilter(
```

Comment:
Why are these commented out?

Reply:
Because it's future functionality not yet available in the Tableau embedded API.